src/sdk/utils/math.ts

```diff
@@ -0,0 +1,45 @@
+import BigNumber from "bignumber.js"
+import { Params } from "src/protojs/nibiru/inflation/v1/genesis"
+
+export const polynomial = (factors: string[], x: BigNumber) => {
+  let result = BigNumber(0)
+  for (let i = 0; i < factors.length; i++) {
+    result = result.plus(
+      BigNumber(factors[i]).times(x.pow(factors.length - i - 1))
+    )
+  }
+
+  return result
+}
+
+export const calculateEpochMintProvision = (
```

Comment:
I'd honestly prefer to just fetch this from the chain, instead of replicating all the math locally, because the period has to be fetched from the chain regardless